notes-with-indexeddb/notes.js

```diff
@@ -0,0 +1,169 @@
+/*
+Feature detection
+ */
+// DON'T use 'var indexedDB = ...' if you're not in a function.
+window.indexedDB = window.indexedDB || window.mozIndexedDB || window.webkitIndexedDB || window.msIndexedDB;
+
+// This line should only be needed if it is needed to support the object's constants for older browsers
+window.IDBTransaction = window.IDBTransaction || window.webkitIDBTransaction || window.msIDBTransaction || {
+    READ_WRITE: 'readwrite'
+};
+
+window.IDBKeyRange = window.IDBKeyRange || window.webkitIDBKeyRange || window.msIDBKeyRange;
+
+if (!window.indexedDB) {
+    window.alert("Your browser doesn't support a stable version of IndexedDB.");
+}
+
+/*
+showErrorMessage from event if available
+ */
```

Comment:
It would be nice to have complete [JSDoc](http://usejsdoc.org/about-getting-started.html) comments for all of these functions, but I won't require it.